spicelib/editor/spice_editor.py

```diff
@@ -756,7 +758,8 @@ def add_instruction(self, instruction: str) -> None:
                 else:
                     i += 1
         elif get_line_command(instruction) == '.PARAM':
-            raise RuntimeError('The .PARAM instruction should be added using the "set_parameter" method')
+            # raise RuntimeError('The .PARAM instruction should be added using the "set_parameter" method')
```

Comment:
I agree. I guess you want to improve the experience of building a netlist using a new set of methods. That's definitely a good idea!

However, this set of methods may need further improvement in the current stage. Some people may even feel unwilling to learn new things. 
If you allow users to build a netlist with both your method to simplify works and spice original instructions as an alternative.

**I have a compromise idea**: 
The spicelib parses instructions received by `add_instructions()`. Instead of directly appending to the netlist lines list, it identifies the ".param" keyword and then calls `set_params()` to add this command and for syntax checking.
The same idea applies to other methods like `set_component` and so on.

Reply:
I agree not to make a change currently.